examples/splinterdb_intro_example.c

```diff
@@ -0,0 +1,135 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * SplinterDB "Hello World" Example Program.
+ */
+#include <stdio.h>
+#include <string.h>
+
+#include "splinterdb/default_data_config.h"
+#include "splinterdb/splinterdb.h"
+#include "example_common.h"
+
+/* Tag to identify messages from application program */
+#define APP_ME "App-Basic"
+
+/*
+ * App-specific 'defaults' that can be parameterized, eventually.
+ */
+#define APP_DB_NAME "splinterdb_intro_db"
+
+#define APP_DEVICE_SIZE_MB 1024 // Size of SplinterDB device; Fixed when created
+#define APP_CACHE_SIZE_MB  64   // Size of cache; can be changed across boots
+
+/* Application declares the limit of key-sizes it intends to use */
+#define APP_MAX_KEY_SIZE ((int)100)
+
+/*
+ * -------------------------------------------------------------------------------
+ * main() Driver for basic SplinterDB example program.
+ * We, intentionally, do not check or show-for error handling, as this is mostly
+ * a sample program to illustrative how-to use the APIs.
+ * -------------------------------------------------------------------------------
+ */
+int
+main()
+{
+   printf("     **** SplinterDB Basic Example program ****\n\n");
+
+   /*
+    * Initialize data configuration, using default key-comparison handling.
+    */
+   data_config splinter_data_cfg;
+   default_data_config_init(APP_MAX_KEY_SIZE, &splinter_data_cfg);
+
+   /* Basic configuration of a SplinterDB instance */
+   splinterdb_config splinterdb_cfg;
+   memset(&splinterdb_cfg, 0, sizeof(splinterdb_cfg));
+   splinterdb_cfg.filename   = APP_DB_NAME;
+   splinterdb_cfg.disk_size  = (APP_DEVICE_SIZE_MB * K_MiB);
+   splinterdb_cfg.cache_size = (APP_CACHE_SIZE_MB * K_MiB);
+   splinterdb_cfg.data_cfg   = &splinter_data_cfg;
+
+   splinterdb *spl_handle = NULL; // To a running SplinterDB instance
+
+   int rc = splinterdb_create(&splinterdb_cfg, &spl_handle);
+   ex_msg("Created SplinterDB instance, dbname '%s'.\n\n", APP_DB_NAME);
+
+   /* Insert a few kv-pairs, describing properties of fruits. */
+   const char *fruit = "apple";
+   const char *descr = "An apple a day keeps the doctor away!";
+   slice       key   = slice_create((size_t)strlen(fruit), fruit);
+   slice       value = slice_create((size_t)strlen(descr), descr);
+   rc                = splinterdb_insert(spl_handle, key, value);
+   ex_msg("Inserted key '%s'\n", fruit);
+
+   fruit = "Orange";
+   descr = "Is a good source of vitamin-C.";
+   key   = slice_create((size_t)strlen(fruit), fruit);
```

Comment:
This is a comment for us to think about the API but this `slice_create` seems like an overkill for many scenarios.